lib/recurly/client.rb

```diff
@@ -166,11 +177,21 @@ def set_headers(request, additional_headers = {})
       request["Accept"] = "application/vnd.recurly.#{api_version}".chomp # got this method from operations.rb
       request["Authorization"] = "Basic #{Base64.encode64(@api_key)}".chomp
       request["User-Agent"] = "Recurly/#{VERSION}; #{RUBY_DESCRIPTION}"
+      request["Idempotency-Key"] ||= generate_idempotency_key unless request.is_a?(Net::HTTP::Get)
```

Comment:
Do we need the keys for DELETEs and HEADs?

Reply:
I changed it so it does not send on GETs and HEADs. DELETEs are a funny one; we have some endpoints that trigger refunds via a delete request. So, we're allowing the idempotency header on those requests.